addons/addon-base-raas/packages/base-raas-services/lib/user/user-service.js

```diff
@@ -90,14 +110,32 @@ class UserService extends BaseUserService {
         errorCount += 1;
       }
     };
+    // Check to make sure users to create don't already exist (if so, exit early with badRequest)
+    await processInBatches(users, batchSize, checkIfUserExists);
+    if (!_.isEmpty(validationErrors)) {
+      throw this.boom
+        .badRequest(`Error: Some specified users already exist. No users were added.`, true)
+        .withPayload(validationErrors);
+    }
+
     // Create users in parallel in the specified batches
     await processInBatches(users, batchSize, createUser);
     if (!_.isEmpty(errors)) {
-      throw this.boom.internalError(`Errors creating users in bulk`, true).withPayload(errors);
+      // Write audit event before throwing error since some users were still added
+      await this.audit(requestContext, {
+        action: 'create-users-batch',
+        body: { totalUsers: _.size(users), completedSuccessfully: false, numErrors: errors },
```

Comment:
`numErrors: errorCount` ? Can we add numSuccess as well? Right now it can be calculated, but it's good to write it out.

Reply:
yep 😅 I fixed this in a later commit, I guess I haven't pushed it yet. Sure, numSuccess can be added as well.